apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.ts

```diff
@@ -134,6 +134,17 @@ GoogleBlockly.Extensions.registerMutator(
   behaviorDefMutator
 );
 
+// This extension make the NAME fields of behavior getter blocks serializable.
+GoogleBlockly.Extensions.register(
```

Comment:
for my understanding: why do we need this for behaviors but not functions? You said in your description it's redundant information for functions, is it not redundant for behaviors?

Reply:
Great question. I had to look into it. The answer is we don't, but we used to. When we did the work to serialize these fields, it was to get us closer to parity with the XML structure CDO Blockly would create. As things work on staging now, you can't embed a behavior call block without a field element because we don't know where to get the name value from. This results in a behavior block without any text. Let's compare XML across Blockly versions:

CDO:
```
<block type="gamelab_behavior_get">
   <mutation />
   <field name="VAR" id="growing">growing</field>
</block>
```
The id attribute above is the behavior id and the text content is the behavior name.

Staging:
```
<block type="gamelab_behavior_get">
    <mutation behaviorId="growing" name="growing"/>
    <field name="NAME">growing</field>
</block>
```
The behavior name shows up in two places above.

At an earlier point in time, we weren't storing the behavior name in the mutation, so it wasn't redundant. Also, with either serialization, we could look at the `textContent` of the field element to find it. In the interest of simplifying and modernizing, I'm going to update this again so we stop serializing these fields. I will have to modify the deserialization hook `domToMutation` to find the name in one of these places:
1. name attribute of the mutation (new)
2. text content of the field (old)
3. use `''` only if neither can be found

I've tested locally that this works well, including with translations.